woocommerce/woocommerce
# CYS - Add tracking for patterns

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/48950



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install the `WooCommerce Beta Tester` plugin.
2. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper`, click the `Features` tab, and enable the `pattern-toolkit-full-composability` feature flag.
3. Install `WooCommerce` from [woocommerce.zip](https://github.com/user-attachments/files/16245982/woocommerce.zip)
4. Go to `wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`.
5. Click on `Design your homepage`.
6. Open the network tools.
7. Search `t.gif` in the network tools filter.
8. Check that you see an event triggered in the network tools for each one of these cases:

| Action | Event |
|--------|--------|
| Click on a category (Intro, About, etc.)  | `customize_your_store_assembler_pattern_category_click` |
| Click on a pattern in the sidebar | `customize_your_store_assembler_pattern_sidebar_click` |
| Close a category panel | `customize_your_store_assembler_pattern_sidebar_close` |
| Click the shuffle button in an inserted pattern | `customize_your_store_assembler_pattern_shuffle_click` | 
| Click the delete button in an inserted pattern | `customize_your_store_assembler_pattern_delete_click` | 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
CYS - Add tracking events to pattern features.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
